### ISSUE TITLE
Creating tuples might be a bottleneck

### ISSUE THREAD

**Issue by KristofferC** [JuliaDebug/JuliaInterpreter.jl#314]
Playing around with PProf.jl it seems we spend a lot of time creating tuples (If I read the output correctly, 20% of full time on summer benchmark):

<img width="691" alt="image" src="https://user-images.githubusercontent.com/1282691/62215136-a7a40c80-b3a6-11e9-82dc-d5b6dc6ab91e.png">

Perhaps it could be optimized somehow.

**Comment by NHDaly** [comment]
>  (If I read the output correctly, 20% of full time on summer benchmark):

Yep! To clarify, that's 0.43% of total time spent actually executing that frame (exclusive), and 20% of total time spent executing this frame and any functions it calls (inclusive)

**Comment by KristofferC** [comment]
This is not so strange since extremely hot functions like `iterate` returns a tuple and we currently never elide forming that.

**Comment by aviatesk** [comment]
I revisited this with a simple `summer` loop on Julia 1.13.0 / Apple M2 Pro.

A relevant improvement is #770 (`fea5d5e6`), which removes the temporary `(fi.framecode, fi.sparam_vals)` tuple previously constructed on each interpreted dispatch-cache hit. The interpreter now returns the cached `FrameInstance` directly.

Comparing that commit against its immediate parent (`3755f47a`), with 10,000-element inputs:

| Input | Time before → after | Allocated memory before → after |
|---|---:|---:|
| `Vector{Int64}` | 109.6 → 98.0 ms | 30.756 → 24.957 MiB |
| `Vector{Float64}` | 111.9 → 99.7 ms | 31.063 → 25.264 MiB |

(For the integer case, the reduction is exactly 190,031 allocations / 6,080,992 bytes, matching 190,031 × 32-byte dispatch-result tuples.)

Separately, profiling current master (`89d3f3a8`) shows that the builtin tuple-construction path through `ntupleany` accounts for approximately 1.3% of CPU samples for both input types. This includes constructing tuples in the interpreted program. So it is distinct from the interpreter-internal dispatch tuples removed by #770.

The original 2019 profile is not directly comparable, so these results do not establish which changes account for the difference from the reported ~20%. But they do confirm that #770 substantially reduces tuple-allocation overhead, and that builtin tuple construction is not a dominant hotspot in this workload today.
